lib/oli/authoring/editing/page_editor.ex

```diff
@@ -203,7 +203,12 @@ defmodule Oli.Authoring.Editing.PageEditor do
            Publishing.get_published_objective_details(publication.id) |> trap_nil(),
          {:ok, objectives_with_parent_reference} <-
            construct_parent_references(objectives) |> trap_nil(),
-         {:ok, tags} <- Oli.Authoring.Editing.TagEditor.list(project_slug, author),
+         {:ok, tags} <-
+           Oli.Authoring.Editing.ResourceEditor.list(
```

Comment:
Refactored the "TagEditor" into a more general "ResourceEditor" that can take a generic resource and do a limited set of operations which can be performed on any resource. This allowed me to reuse this module for alternatives edits, as well as any future version-able resources we might need in the future as well